Add a cloneUrl function which will git clone a repo instead of git init it

mbontemps opened the pull request:
I think we need that as `git clone` does a little more than a `git init`, `git remote add`, `git pull origin master`: it discovers the main branch without the need to know its name (useful if it's `develop` and not `master` for example).

mbontemps committed:
Add a cloneUrl function which will git clone a repo instead of git init it

ornicar commented on lib/PHPGit/Repository.php:
```diff
@@ -76,6 +76,28 @@ class PHPGit_Repository
+     * @param   array $options
+     * @return PHPGit_Repository
+     **/
+    public static function cloneUrl($dir, $url, $debug = false, array $options = array())
```
that's a good one, thanks. But I'd like to discuss the function signature. It would be closer to the Git API to have something like:

```
function cloneUrl($url, $dir = null, ...)
```

where dir can be inferred from the url, just like Git does:

```
git clone https://github.com/foo/bar # clones to bar/

git clone https://github.com/foo/bar baz # clones to baz/
```

mbontemps replied:
Agreed that inverting 1st and 2nd params would feel more natural.

About allowing `dir = null`, you mean like using `getcwd`? Not sure this is something lots of people use…

ornicar replied:
Agreed! Let's invert params but keep dir required.

mbontemps committed:
Change parameter orders in cloneUrl

mbontemps committed:
Execute cloneUrl from current directory

This allows cloning a repo in a dir does not yet exist.
(The dir will be created by git)

ornicar commented:
better late than never, as they say.
